elastic/kibana
# Config to toggle Custom Roles

## Summary

Closes https://github.com/elastic/kibana/issues/174771

While the security team works on Custom Roles for serverless, we want to hide the content behind a feature flag.

An existing config option that was used to hide the Roles UI during the initial phases of serverless has been repurposed, and will now toggle both the Roles UI and the Roles Routes

`xpack.security.confg.ui.roleManagementEnabled` has been changed to `xpack.security.confg.roleManagementEnabled` and will have to be set to `true` in a config file while in serverless mode to show the Roles card on the management screen and enable the UI/routes.

## Reviewers

Ive included a `viewer`:`changeme` user for testing (It will be removed after approval).

## Testing

### xpack.security.config.roleManagementEnabled

1. In your `kibana.yml`, add `xpack.security.confg.roleManagementEnabled: true`
2. Start up in serverless mode locally, login in with `elastic_serverless`:`changeme`
3. Click `Project Settings` > `Management`
4. `Roles` card should display under `Other`
5. Navigate to `Roles`, it displays, but the `Edit Roles` page does not work yet.

### Test as Viewer

1. In your `kibana.yml`, add either above option as you prefer 
2. Start up in serverless mode locally, login in with `viewer`:`changeme`
3. Click `Project Settings` > `Management`
4. `Roles` card should NOT display under `Other` and the `roles` URL should not work.

## Screenshots

Roles card
<img width="1281" alt="Screenshot 2024-02-05 at 3 22 12 PM" src="https://github.com/elastic/kibana/assets/21210601/a1285ada-7ff7-495f-88a6-9847b3245518">
